packages/custom-props-layer/src/components/text/Text.tsx

```diff
@@ -3,19 +3,34 @@ import { useMemo, useCallback } from "react";
 import { Label } from "../commons/Label";
 import { PropertyContainer } from "../commons/PropertyContainer";
 import { TextInput } from "../commons/TextInput";
+
 export const Text: React.FC<ComponentProps> = (props) => {
   const propValue = useMemo(() => {
+    if (props.objectName)
+      return props.customProps[props.objectName][props.propName] || "";
     return props.customProps[props.propName] || "";
   }, [props]);
   const callPatchCb = useCallback(
     (e: React.ChangeEvent<HTMLInputElement>) => {
-      props.patchCb({
-        property: {
-          custom: {
-            [props.propName]: e.target.value,
+      if (props.objectName) {
+        props.patchCb({
+          property: {
+            custom: {
+              [props.objectName]: {
```

Comment:
To make this part generic, we the `Text` component here should take the following prop:

```tsx
type TextProp = {
 selector: string[], // replace propName with this new field
}
```

Then before calling patch, you need to loop over this `selector` array and create an object to pass on to `patchCb` function something like below:

```tsx
props.patchCb({
 property: {
  custom: objectCreatedAfterLoopingOverSelectorArray
 }
})
```

Example: If the selector is `["a", "b",  "c"]`, then the object will be like:

```tsx
const objectCreatedAfterLoopingOverSelectorArray = {
 a: {
  b: {
   c: e.target.value
  }
 }
}
```

Reply:
Done